### DISCUSSION TITLE
Text Panel: Support render HTML/markdown using templating of data source query response

### DISCUSSION THREAD

**Discussion by pachico** [grafana/grafana#38358] [Dashboards, Panels & Visualizations]
Many times it is necessary to display in Dashboards the content of what in a database would be the columns of a single row: details of a booking, of a node, last point statistics, etc.To do this, we can only use the table panel and put the values one after the other in columnar fashion.This is ok but the space usage and readability are not great.
My proposal is to have panel where the content is pure HTML with placeholders that will be interpolated with the values of a single row.Ideally, it could be a repeatable panel, one for each row returned by the data source.
![image](https://user-images.githubusercontent.com/2184438/73248595-884e3680-41b3-11ea-8acb-43a6542560dc.png)
![image](https://user-images.githubusercontent.com/2184438/73248636-956b2580-41b3-11ea-899e-fd64ae3ba9e2.png)

Hope you find this interesting.
Cheers

**Comment by marefr** [comment]
You should be able to do this today in text panel using template variables. Let me know if that doesn't satisfy this feature request.

**Comment by pachico** [comment]
Thanks for the quick reply, @marefr.
I understand you could interpolate template variables but not data coming from data source  (unless template variables were coming from data source themselves, but this would only make a longer round trip to get the data).
I might want to have many pieces of data that would force me to have as many template variables, which isn't the case.

What I mean is, from one variable (id of any sort, template or not), query data source, retrieve row data, interpolate with template. For that, the Text panel would certainly require a data source which currently doesn't. 
As an extra feature, as I mentioned, this could be a repeating panel, one per row found in data source.

**Reply by AbelThorne** [reply]
You are right, in the current text panel, you can only use template variables, not results from the queries. 
There is a third-party plugin that will do that for you: [Dynamic Text](https://grafana.com/grafana/plugins/marcusolsson-dynamictext-panel/). However I agree that this should be integrated into the core Text panel as this is really useful

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82993. Feel free to continue the discussion around this there. Thank you!
